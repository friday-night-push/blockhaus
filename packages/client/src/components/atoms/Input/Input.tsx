import type { TextInputProps } from '@gravity-ui/uikit';
import { TextInput } from '@gravity-ui/uikit';

export type InputProps = {
  name: string;
  required?: boolean;
} & TextInputProps;

export const Input = ({
  label,
  name,
  type = 'text',
  onChange,
  ...props
}: InputProps) => {
  return (
    <TextInput
      label={label}
      name={name}
      type={type}
      onChange={onChange}
      size='xl'
      view='normal'
      pin='brick-brick'
      {...props}
    />
  );
};
