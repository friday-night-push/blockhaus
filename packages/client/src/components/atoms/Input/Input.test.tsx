import { render } from '@testing-library/react';

import { Input } from './Input';

describe('Input', () => {
  it('renders correctly with default props', () => {
    const { container } = render(<Input name={'test'} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
