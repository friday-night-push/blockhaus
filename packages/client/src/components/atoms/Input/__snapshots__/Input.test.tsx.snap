// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly with default props 1`] = `
<span
  class="g-text-input g-text-input_view_normal g-text-input_size_xl g-text-input_pin_brick-brick"
>
  <span
    class="g-text-input__content"
  >
    <input
      class="g-text-input__control g-text-input__control_type_input"
      name="test"
      type="text"
      value=""
    />
  </span>
</span>
`;
